src/grlib.c

```diff
@@ -1019,217 +1013,263 @@ apbuart_restore_stdio (void)
 #define DO_STDIO_READ( _fd_, _buf_, _len_ )          \
 		( dumbio || nouartrx ? (0) : read( _fd_, _buf_, _len_ ) )
 
-static void
-apbuart_init (void)
+int
+uart_init(UART_DEF *uart)
 {
-  f1in = stdin;
-  f1out = stdout;
-  if (uart_dev1[0] != 0)
+  int result = 0;
+
+  uart->in_stream.file = stdin;
```

Comment:
This will be changed for multiple UARTS,right? Only one can use stdio...

Reply:
yes